### ISSUE TITLE
[Tearsheet]: Focus not trapping inside Tearsheet untill any element in the tearsheet is selected

### ISSUE THREAD

**Issue by Geethkadali** [carbon-design-system/ibm-products#5590]
### Package

@carbon/ibm-products

### Browser

Chrome, Safari, Firefox, Edge

### Package version

V2.43.1

### React version

v18.3.1

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

Focus state should not leave the tear sheet when enabled. Currently, the focus starts on the tearsheet until it gets to the action buttons, goes outside, then eventually comes back to the “x” on the tearsheet to close it. If we select any element in the Tearsheet then the focus is trapping inside and not going outside.

https://github.com/carbon-design-system/ibm-products/assets/45656474/330eb270-a3e4-48bc-9a78-f583c5e0e2d6

### Reproduction/example

https://stackblitz.com/edit/vitejs-vite-1moieg?file=src%2FApp.jsx

### Steps to reproduce

Interact with the tear sheet using keyboard. 
Focus is leaving the tearsheet and going outside, only when we interact with the tearsheet first by using "enter" on one of the options then the focus is staying inside.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by elycheea** [comment]
As discussed over Slack, this appears to be an issue with the implementation and not the Tearsheet itself.

Would like to reiterate that the `CreateTearsheet` is designed with these stepped flows in mind, but we would need to better understand the justification for removing the influencer that shows the current steps.
